patches/minecraft/net/minecraft/client/util/RecipeBookCategories.java.patch

```diff
@@ -0,0 +1,188 @@
++               break;
++            case EQUIPMENT:
++               category.setupFilter(net.minecraft.item.crafting.IRecipeType.field_222149_a, RecipeBookCategories::equipmentPredicate);
++               break;
++            case MISC:
++               category.setupFilter(net.minecraft.item.crafting.IRecipeType.field_222149_a, RecipeBookCategories::miscPredicate);
++               break;
++            case FURNACE_SEARCH:
++               category.setupSearch(net.minecraft.item.crafting.IRecipeType.field_222150_b);
++               break;
++            case FURNACE_FOOD:
++               category.setupFilter(net.minecraft.item.crafting.IRecipeType.field_222150_b, RecipeBookCategories::foodPredicate);
++               break;
++            case FURNACE_BLOCKS:
++               category.setupFilter(net.minecraft.item.crafting.IRecipeType.field_222150_b, RecipeBookCategories::furnaceBlocksPredicate);
++               break;
++            case FURNACE_MISC:
++               category.setupFilter(net.minecraft.item.crafting.IRecipeType.field_222150_b, RecipeBookCategories::furnaceMiscPredicate);
++               break;
++            case BLAST_FURNACE_SEARCH:
++               category.setupSearch(net.minecraft.item.crafting.IRecipeType.field_222151_c);
++               break;
++            case BLAST_FURNACE_BLOCKS:
++               category.setupFilter(net.minecraft.item.crafting.IRecipeType.field_222151_c, RecipeBookCategories::blastFurnaceBlocksPredicate);
++               break;
++            case BLAST_FURNACE_MISC:
++               category.setupFilter(net.minecraft.item.crafting.IRecipeType.field_222151_c, RecipeBookCategories::blastFurnaceMiscPredicate);
++               break;
++            case SMOKER_SEARCH:
++               category.setupSearch(net.minecraft.item.crafting.IRecipeType.field_222152_d);
++               break;
++            case SMOKER_FOOD:
++               category.setupFilter(net.minecraft.item.crafting.IRecipeType.field_222152_d, RecipeBookCategories::foodPredicate);
++               break;
++            case STONECUTTER:
++               category.setupSearch(net.minecraft.item.crafting.IRecipeType.field_222154_f);
++               break;
++            case CAMPFIRE:
++               category.setupSearch(net.minecraft.item.crafting.IRecipeType.field_222153_e);
++         }
++
++         register(category, category.name().toLowerCase());
++      }
++   }
++
++   @Override
++   public final RecipeBookCategories setRegistryName(net.minecraft.util.ResourceLocation name) {
++      return setRegistryName(name.toString());
++   }
++
++   @javax.annotation.Nullable
++   @Override
++   public net.minecraft.util.ResourceLocation getRegistryName() {
++      return registryName;
++   }
++
++   @Override
++   public Class<net.minecraftforge.common.extensions.IForgeRecipeBookCategory<? extends net.minecraft.item.crafting.IRecipeType<?>>> getRegistryType() {
++      //return RecipeBookCategories.class;
```

Comment:
I am not sure returning null here is a good idea, especially since this goes into an `IForgeRegistry`but i could be wrong here.
Anybody with more information on that system should be able to tell you if this is correct or not.

Reply:
Added missing return value